code/game/gamemodes/changeling/powers/adrenaline.dm

```diff
@@ -31,10 +31,19 @@
 	overdose_threshold = 30
 
 /datum/reagent/medicine/changelingAdrenaline/on_mob_life(mob/living/M as mob)
-	M.AdjustParalysis(-1)
-	M.AdjustStunned(-1)
-	M.AdjustWeakened(-1)
-	M.adjustStaminaLoss(-1)
+	if(!(M.stunned || M.weakened || M.paralysis))
```

Comment:
To check if they aren't stunned before increasing the timer, as well as resetting the metabolism.